### PR TITLE
Update from upstream

### PR DESCRIPTION
**Pull request by 0xDCA**
This PR updates this fork with the latest changes from UCL-INGI/INGInious

### CODE REVIEW

File: inginious/frontend/webapp/pages/course_admin/task_edit.py [hunk 1]
```diff
@@ -268,10 +267,16 @@ def postprocess_grader_data(self, data, directory_path):
                 return json.dumps(
                     {"status": "error", "message": "You must provide test cases to autogenerate the grader"})
 
-            with open(target_run_file, "w") as f:
-                f.write(run_file_template.format(
-                    problem_id=repr(problem_id), test_cases=repr(test_cases),
-                    options=repr(options), weights=repr(weights)))
+            with tempfile.TemporaryDirectory() as temporary_folder_name:
```
Comment (amrondonp) [inline inginious/frontend/webapp/pages/course_admin/task_edit.py:270]: @0xDCA Could you explain what are you doing here with temporary directories?

File: inginious/frontend/webapp/pages/course_admin/task_edit_file.py [hunk 2]
```diff
@@ -274,19 +260,19 @@ def action_list_as_json(self, courseid, taskid):
...
-                "is_directory": is_directory,
-                "name": name,
-                "complete_name": complete_name
-            } for level, is_directory, name, complete_name in file_list
+              {
+              "level": level,
+              "is_directory": is_directory,
+              "name": name,
+              "complete_name": complete_name
+              } for level, is_directory, name, complete_name in file_list
```
Comment (amrondonp) [inline inginious/frontend/webapp/pages/course_admin/task_edit_file.py:275]: @0xDCA Did you only make corrections style in this file?

File: inginious/frontend/webapp/templates/task.html [hunk 3]
```diff
@@ -223,7 +225,7 @@ <h4 class="modal-title">Output diff</h4>
         </div>
     </div>
 
-<form id="task" method="post" action="$get_homepath()/course/${task.get_course().get_id()}/${task.get_id()}" enctype="multipart/form-data"
+<form id="task" method="post" enctype="multipart/form-data"
```
Comment (amrondonp) [inline inginious/frontend/webapp/templates/task.html:228]: @0xDCA Do you know how the submissions work now? Maybe was due to this that mine's were failing?

File: inginious/frontend/webapp/pages/course_admin/task_edit.py (hunk 1 above)
Comment (0xDCA) [reply inginious/frontend/webapp/pages/course_admin/task_edit.py:270]: I'm using the new Inginious' [file system API](https://github.com/UCL-INGI/INGInious/blob/master/inginious/common/filesystems/provider.py) to write the run file into the task folder. There is no method to copy a single file, but there is a method to copy the contents of a folder, so I'm temporarily writing the run file to a temporary directory, and copying the contents of that directory to the task folder. There is a put() method, but I find it a bit underspecified in terms of the format of the contents parameter (not sure if it's binary or if it assumes a specific encoding)

File: inginious/frontend/webapp/pages/course_admin/task_edit_file.py (hunk 2 above)
Comment (0xDCA) [reply inginious/frontend/webapp/pages/course_admin/task_edit_file.py:275]: No, I took the new file and re-applied the following modifications:
- Method for listing the task files as JSON
- Customization of message so the error is less cryptic when we add a file to a task that hasn't been created yet.

File: inginious/frontend/webapp/templates/task.html (hunk 3 above)
Comment (0xDCA) [reply inginious/frontend/webapp/templates/task.html:228]: It's the same URL as before, they just removed this URL as they found it redundant.

File: inginious/frontend/webapp/pages/course_admin/task_edit_file.py (hunk 2 above)
Comment (amrondonp) [reply inginious/frontend/webapp/pages/course_admin/task_edit_file.py:275]: @0xDCA I has been very difficult for me to find the changes you've made. Could you please put a set of range of lines? Thank you

**Pull request merged**